src/classes/BGE_BatchGiftEntry_UTIL.cls

```diff
@@ -102,4 +102,40 @@ public with sharing class BGE_BatchGiftEntry_UTIL {
         };
     }
 
+    /*******************************************************************************************************
+    * @description Returns the source field's options formatted for lightning:select if the source field is a picklist
+    * or the target field's options formatted for lightning:select if the source field is a string and the target
+    * field is a picklist
+    * Returns an empty map if neither case applies
+    */
+    public static List<Map<String, String>> getPicklistOptions(String dataImportField){
+        List<Map<String, String>> options = new List<Map<String, String>>();
+
+        String dataImportObject = UTIL_Namespace.StrTokenNSPrefix('DataImport__c');
+
+        String rawSourceFieldType = UTIL_Describe.getFieldDisplaytype(dataImportObject, dataImportField.toLowerCase()).name().toLowerCase();
+
+        if (rawSourceFieldType == 'picklist') {
+            options = UTIL_Describe.getLightningSelectOptions(dataImportObject, dataImportField);
+
+        } else if (rawSourceFieldType == 'string') {
+
+            String targetObject = BDI_DataImportService.getTargetObject(dataImportField);
+            if (targetObject == 'Payment') {
+                targetObject = 'npe01__OppPayment__c';
+            }
+
+            String targetField = BDI_DataImportService.getTargetField(dataImportField);
+            Schema.DisplayType targetFieldType = UTIL_Describe.getFieldDisplaytype(targetObject, targetField);
+
+            if (targetFieldType == Schema.DisplayType.PICKLIST) {
+                options = UTIL_Describe.getLightningSelectOptions(targetObject, targetField);
+            } else if (targetField == 'RecordTypeId') {
+                options = UTIL_RecordTypes.getStringRecordTypesForLightningSelect(UTIL_Describe.getSObjectType(targetObject));
```

Comment:
This block looks like a big change, but really it was just moved + I added this little block to handle additional options!